src/api/entities/MultiSig/MultiSigProposal.ts

```diff
@@ -0,0 +1,131 @@
+import { BigNumber } from 'bignumber.js';
+
+import { Context, Entity, PolymeshError } from '~/internal';
+import { ErrorCode, MultiSigProposalDetails, TxTag } from '~/types';
+import {
+  bigNumberToU64,
+  boolToBoolean,
+  meshProposalStatusToProposalStatus,
+  momentToDate,
+  stringToAccountId,
+  u64ToBigNumber,
+} from '~/utils/conversion';
+import { assertAddressValid } from '~/utils/internal';
+
+interface UniqueIdentifiers {
+  multiSigAddress: string;
+  id: BigNumber;
+}
+
+export interface HumanReadable {
+  multiSigAddress: string;
+  id: string;
+}
+
+/**
+ * A proposal for a MultiSig transaction. This is a wrapper around an extrinsic that will be executed when the amount of approvals reaches the signature threshold set on the MultiSig Account
+ */
+export class MultiSigProposal extends Entity<UniqueIdentifiers, HumanReadable> {
+  public multiSigAddress: string;
+  public id: BigNumber;
+
+  /**
+   * @hidden
+   */
+  constructor(identifiers: UniqueIdentifiers, context: Context) {
+    super(identifiers, context);
+    const { multiSigAddress, id } = identifiers;
+    assertAddressValid(multiSigAddress, context.ss58Format);
+    this.multiSigAddress = multiSigAddress;
+    this.id = id;
+  }
+
+  /**
+   * Fetches the details of the Proposal. This includes the amount of approvals and rejections, the expiry, and details of the wrapped extrinsic
+   */
+  public async details(): Promise<MultiSigProposalDetails> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { multiSig },
+        },
+      },
+      multiSigAddress,
+      id,
+      context,
+    } = this;
+    const rawMultiSignAddress = stringToAccountId(multiSigAddress, context);
+    const rawId = bigNumberToU64(id, context);
+    const [
+      {
+        approvals: rawApprovals,
+        rejections: rawRejections,
+        status: rawStatus,
+        expiry: rawExpiry,
+        autoClose: rawAutoClose,
+      },
+      proposal,
+    ] = await Promise.all([
+      multiSig.proposalDetail([rawMultiSignAddress, rawId]),
+      multiSig.proposals([rawMultiSignAddress, rawId]),
+    ]);
```

Comment:
I left this as Promise.all since the types and code is cleaner and the extra overhead should be minimal.

Thoughts @prashantasdeveloper ? I left a similar comment on asset metadata PR. Maybe we should have a rule like more than 3 calls need to go to multi.

I'd be more open to using multi more if we didn't have to assert all the types.